src/mainClass.js

```diff
@@ -0,0 +1,107 @@
+import { getData, getAllCategories } from "./source";
+
+export class AllItems {
+
+  setData(searchCategory, searchLinks, displayLinksForChosenCategory, searchChosenLinksOnly) {
+    this.searchThroughCategories = searchCategory;
+    this.searchThroughLinks = searchLinks;
+    this.displayChosenLinks = displayLinksForChosenCategory;
+    this.categories = [];
+    this.filteredCategoriesAfterSearch = [];
+    this.links = [];
```

Comment:
filteredLinksAfterSearch and filteredCategoriesAfterSearch not usless props, don't create props which are basicly local variables